### ISSUE TITLE
Note: computing entropy on constraint system

### ISSUE THREAD

**Issue by tanlin2013** [tanlin2013/qlinks#40]
* Consider these bases for a spin-1/2 chain of length 3

| index / site | 0 | 1 | 2 |
|------------|---|---|---|
| 0          | 0 | 0 | 0 |
| 1          | 0 | 0 | 1 |
| 2          | 0 | 1 | 0 |
| 4          | 1 | 0 | 0 |
| 5          | 1 | 0 | 1 |

---

* If the bipartition of sites is $A = \\{ 0, 1 \\}$ and $B = 2$, the bases can be sorted accordingly to

| index / site | 0 | 1 | 2 |
|------------|---|---|---|
| 0          | 0 | 0 | 0 |
| 2          | 0 | 1 | 0 |
| 4          | 1 | 0 | 0 |
| 1          | 0 | 0 | 1 |
| 5          | 1 | 0 | 1 |

for partition $A$, each basis has index

$\ket{00} \rightarrow 0$
$\ket{01} \rightarrow 1$
$\ket{10} \rightarrow 2$

Similarly, for partition $B$, there are only 2 indices  0 and 1

---

* This can be reorganized into

| B index / A index| 0 | 1 | 2 |
|-----|---|---|---|
| 0   | 0 | 2 | 4 |
| 1   | 1 | N/A  | 5 |

where each cell is the original index

---

* In short,

$$
\ket{\psi} = \sum_k c_k \ket{e_k} = \sum_{i_A}\sum_{i_B} \chi_{i_A, i_B} \ket{e_{i_A}^{(A)}} \otimes \ket{e_{i_B}^{(B)}}
$$

This is how the matrix $\chi$ can be constructed.

Ref
---
See the appendix of this article https://link.aps.org/doi/10.1103/PhysRevLett.126.220601
